multiqc/modules/pangolin/pangolin.py

```diff
@@ -0,0 +1,164 @@
+#! /usr/bin/env python
+
+""" MultiQC module to parse output from Pangolin """
+
+from __future__ import print_function
+from collections import OrderedDict
+import logging
+import csv
+
+from multiqc import config
+from multiqc.plots import beeswarm, table
+from multiqc.modules.base_module import BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    """Pangolin module"""
+
+    def __init__(self):
+
+        # Initialise the parent module
+        super().__init__(
+            name="Pangolin",
+            anchor="pangolin",
+            href="https://github.com/cov-lineages/pangolin",
+            info="uses variant calls to assign SARS-CoV-2 genome sequences to global lineages.",
+        )
+
+        # Find and parse the sample files
+        self.pangolin_data = dict()
+        for f in self.find_log_files("pangolin", filehandles=True):
+            self.parse_pangolin_log(f)
+
+        # Filter out parsed samples based on sample name
+        self.pangolin_data = self.ignore_samples(self.pangolin_data)
+
+        # Stop if we didn't find anything
+        if len(self.pangolin_data) == 0:
+            raise UserWarning
+
+        self.pangolin_general_stats_table()
+
+        self.add_section(
+            name="Run table",
+            anchor="pangolin-run-table",
+            description="Statistics gathered from the input pangolin files",
+            plot=self.pangolin_table(),
+        )
+
+    def parse_pangolin_log(self, fh):
+        contents = csv.DictReader(fh["f"])
+        for row in contents:
+            taxon_name = row["taxon"]
+            row.pop("taxon")
+            self.pangolin_data[taxon_name] = row
```

Comment:
This is nice, but you're far too trusting. MultiQC is used by enough people in enough weird situations that we need to expect people to throw all kinds of random inputs at it.

```console
$ cat > not_pangolin.csv
This looks like it's from Pangolin (pangolin_version) but it's not! It's a lie!
Not really a csv
No data.
Should break horribly.

$ multiqc -f .

  /// MultiQC 🔍 | v1.11.dev0 (9b5a1d1)

|           multiqc | Search path : /Users/ewels/GitHub/MultiQC/TestData/data/modules/pangolin
|         searching | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 7/7
╭────────────────────────────────── Oops! The 'pangolin' MultiQC module broke... ───────────────────────────────────╮
│ Please copy this log and report it at https://github.com/ewels/MultiQC/issues                                     │
│ Please attach a file that triggers the error. The last file found was: ./not_pangolin.csv                         │
│                                                                                                                   │
│ Traceback (most recent call last):                                                                                │
│   File "/Users/ewels/GitHub/MultiQC/MultiQC/multiqc/multiqc.py", line 604, in run                                 │
│     output = mod()                                                                                                │
│   File "/Users/ewels/GitHub/MultiQC/MultiQC/multiqc/modules/pangolin/pangolin.py", line 34, in __init__           │
│     self.parse_pangolin_log(f)                                                                                    │
│   File "/Users/ewels/GitHub/MultiQC/MultiQC/multiqc/modules/pangolin/pangolin.py", line 55, in parse_pangolin_log │
│     taxon_name = row["taxon"]                                                                                     │
│ KeyError: 'taxon'                                                                                                 │
│                                                                                                                   │
╰───────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
|           multiqc | No analysis results found. Cleaning up..
|           multiqc | MultiQC complete
```

Reply:
You also need to clean the sample name, as described here: https://multiqc.info/docs/#custom-sample-names

And also log about identical sample names, as described here: https://multiqc.info/docs/#identical-sample-names

Fixing all of this in a commit I'm about to push 👍🏻